### PR TITLE
feat: add push provider

### PR DESCRIPTION
**Pull request by peterdeme**
Add push providers

### CODE REVIEW

File: src/main/java/io/getstream/chat/java/models/App.java [hunk 1]
```diff
@@ -791,6 +814,107 @@ public AppGetRateLimitsRequest endpoint(@NotNull String endpoint) {
     }
   }
 
+  @Data
+  @NoArgsConstructor
+  public static class PushProvider {
```
Comment (driver-devel) [inline src/main/java/io/getstream/chat/java/models/App.java:819]: Why the list of fields is not complete ? Are we going to skip legacy fields like templates etc... ?

Comment (driver-devel) [reply src/main/java/io/getstream/chat/java/models/App.java:819]: Ignore me. I should be the same as we do in go-sdk.

Comment (peterdeme) [reply src/main/java/io/getstream/chat/java/models/App.java:819]: yup, same as Go

**Review by driver-devel — APPROVED** [review]
LGTM

File: src/main/java/io/getstream/chat/java/models/App.java [hunk 2]
```diff
@@ -791,6 +814,107 @@ public AppGetRateLimitsRequest endpoint(@NotNull String endpoint) {
...
+    @JsonProperty("apn_team_id")
+    private String apnTeamId;
+
+    @Nullable
+    @JsonProperty("apn_topic")
+    private String apnTopic;
+
+    @Nullable
+    @JsonProperty("firebase_server_key")
+    private String firebaseServerKey;
```
Comment (ferhatelmas) [inline src/main/java/io/getstream/chat/java/models/App.java:858]: let's drop this

**Review by ferhatelmas — CHANGES_REQUESTED** [review]
one small touch is needed, otherwise looks good

File: src/main/java/io/getstream/chat/java/models/App.java (hunk 2 above)
Comment (peterdeme) [reply src/main/java/io/getstream/chat/java/models/App.java]: ✅

**Review by ferhatelmas — APPROVED** [review]

**Pull request merged**